mmdet3d/models/dense_heads/primitive_head.py

```diff
@@ -0,0 +1,816 @@
+                        linecenter[0], linecenter[1], linecenter[2],
+                        (pts_semantic_mask[ind][0])
+                    ]).to(device)
+                if torch.sum(line_sel2) > self.train_cfg['num_point_line']:
+                    point_line_mask[ind[sel][line_sel2]] = 1.0
+                    linecenter = torch.mean(x[sel][line_sel2], axis=0)
+                    linecenter[1] = (ymin + ymax) / 2.0
+                    point_line_offset[
+                        ind[sel][line_sel2]] = linecenter - x[sel][line_sel2]
+                    point_line_sem[ind[sel][line_sel2]] = torch.as_tensor([
+                        linecenter[0], linecenter[1], linecenter[2],
+                        (pts_semantic_mask[ind][0])
+                    ]).to(device)
+                if torch.sum(line_sel3) > self.train_cfg['num_point_line']:
+                    point_line_mask[ind[sel][line_sel3]] = 1.0
+                    linecenter = torch.mean(x[sel][line_sel3], axis=0)
+                    linecenter[0] = (xmin + xmax) / 2.0
+                    point_line_offset[
+                        ind[sel][line_sel3]] = linecenter - x[sel][line_sel3]
+                    point_line_sem[ind[sel][line_sel3]] = torch.as_tensor([
+                        linecenter[0], linecenter[1], linecenter[2],
+                        (pts_semantic_mask[ind][0])
+                    ]).to(device)
+                if torch.sum(line_sel4) > self.train_cfg['num_point_line']:
+                    point_line_mask[ind[sel][line_sel4]] = 1.0
+                    linecenter = torch.mean(x[sel][line_sel4], axis=0)
+                    linecenter[0] = (xmin + xmax) / 2.0
+                    point_line_offset[
+                        ind[sel][line_sel4]] = linecenter - x[sel][line_sel4]
+                    point_line_sem[ind[sel][line_sel4]] = torch.as_tensor([
+                        linecenter[0], linecenter[1], linecenter[2],
+                        (pts_semantic_mask[ind][0])
+                    ]).to(device)
+
+            if self.primitive_mode == 'z':
+                if torch.sum(sel) > self.train_cfg['num_point'] and torch.var(
+                        alldist[sel]) < self.train_cfg['var_thresh']:
+                    center = torch.as_tensor([
+                        (xmin + xmax) / 2.0, (ymin + ymax) / 2.0,
+                        torch.mean(x[sel][:, 2])
+                    ]).to(device)
+                    sel_global = ind[sel]
+                    point_boundary_mask_z[sel_global] = 1.0
+                    point_boundary_sem_z[sel_global] = torch.as_tensor([
+                        center[0], center[1], center[2], xmax - xmin,
+                        ymax - ymin, (pts_semantic_mask[ind][0])
+                    ]).to(device)
+                    point_boundary_offset_z[sel_global] = center - x[sel]
+
+            # Get left two lines
+            v1 = corners[3] - corners[2]
```

Comment:
v1, v2, cp, d, a, b, c are not good variable names. Use more clear names with regarding to their real usages and meaning.